nexus/src/app/sagas/instance_migrate.rs

```diff
@@ -213,33 +140,23 @@ async fn sim_reserve_sled_resources(
 ) -> Result<Uuid, ActionError> {
     let osagactx = sagactx.user_data();
     let params = sagactx.saga_params::<Params>()?;
+    let propolis_id = sagactx.lookup::<Uuid>("dst_propolis_id")?;
 
-    // N.B. This assumes that the instance's shape (CPU/memory allotment) is
-    //      immutable despite being in the instance's "runtime" state.
-    let resources = db::model::Resources::new(
-        params.instance.runtime_state.ncpus.0 .0.into(),
-        params.instance.runtime_state.memory,
-        // TODO(#2804): Properly specify reservoir size.
-        omicron_common::api::external::ByteCount::from(0).into(),
-    );
-
-    // Add a constraint that the only allowed sled is the one specified in the
-    // parameters.
+    // Add a constraint that requires the allocator to reserve on the
+    // migration's destination sled instead of a random sled.
     let constraints = db::model::SledReservationConstraintBuilder::new()
         .must_select_from(&[params.migrate_params.dst_sled_id])
         .build();
 
-    let propolis_id = sagactx.lookup::<Uuid>("dst_propolis_id")?;
-    let resource = osagactx
-        .nexus()
-        .reserve_on_random_sled(
-            propolis_id,
-            db::model::SledResourceKind::Instance,
-            resources,
-            constraints,
-        )
-        .await
-        .map_err(ActionError::action_failed)?;
+    let resource = super::instance_common::reserve_vmm_resources(
```

Comment:
(no action) this is a lot more readable than "reserve_on_random_sled ('please pick exactly this sled')" lol

Reply:
I mean, we still do that. You just don't have to think about it unless you go into that specific file ;)